src/DumpOnException.Dumpster/Listener.cs

```diff
@@ -13,10 +13,35 @@ internal class Listener
         private static DiagnosticsClient? _client;
         private static int _count;
         private static int _running;
+        private static Thread _memoryThresholdThread;
+        private static Process _currentProcess;
     
         public Listener()
         {
             AppDomain.CurrentDomain.FirstChanceException += CurrentDomainOnFirstChanceException;
+
+            if (Settings.MemoryThreshold > 0)
+            {
+                _currentProcess = Process.GetCurrentProcess();
+                _memoryThresholdThread = new Thread(() =>
```

Comment:
You need to set IsBackground to true on the thread, otherwise it will prevent the app from shutting down